### ISSUE TITLE
Appservice forgets nickname after postgres migration

### ISSUE THREAD

**Issue by twouters** [matrix-org/matrix-appservice-irc#961]
UPDATED: noticed that the original server hostname contains underscores, the newly added entry after running `!nick` uses dots instead.

Original nedb database contains the following record for my account:

```json
{
  "type": "matrix",
  "id": "@<localpart>:localhost",
  "data": {
    "client_config": {
      "<server_name_tld>": {
        "nick": "<nickname>",
        "username": "<username>"
      }
    },
    "localpart": "<localpart>"
  },
  "_id": "<mxid>"
}
```

After migrating to postgresql the following client config entry can be found:

```
matrix_appservice_test=# select * from client_config;
           user_id           |    domain    |                    config                     | password 
-----------------------------+--------------+-----------------------------------------------+----------
 @<localpart>:localhost      | <server_name_tld> | {"nick":"<nickname>","username":"<username>"} | 
(1 row)
```

As soon as the service is started, it connects with nickname `<localpart>[m]`

After setting the correct nickname with `!nick <nickname>` I get a second entry in the database:

```
matrix_appservice_test=# select * from client_config;
           user_id           |    domain    |                    config                     | password 
-----------------------------+--------------+-----------------------------------------------+----------
 @<localpart>:localhost      | <server_name_tld> | {"nick":"<nickname>","username":"<username>"} | 
 @<localpart>:localhost      | <server.name.tld> | {"username":"<username>","nick":"<nickname>"} |
(2 rows)
```

**Comment by twouters** [comment]
https://github.com/matrix-org/matrix-appservice-irc/blob/59a68670da2bad0ee6f569a256088fcf5d307154/src/datastore/NedbDataStore.ts#L105

**Comment by twouters** [comment]
I guess this could be solved by replacing `_` with `.` during the migration

**Comment by Half-Shot** [comment]
Already on a PR :)

**Comment by brycied00d** [comment]
I just ran into this issue while running 0.19.0 and migrating from nedb to postgres (yes, finally). Looking at `src/scripts/migrate-db-to-pgres.ts`, the fix from #962 is *not* there. In fact, I can't find any record of #962 in the git history for either `develop` or `master` branches, despite Github indicating that it was merged. Any clue what's happened?

I believe this issue should be reopened until #962 is re-merged.
